src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -146,6 +148,7 @@ public string GetServerUrl(CommandSettings command)
                 _collectionName = command.GetCollectionName();
             }
 
+            _command = command;
```

Comment:
Seems weird you cache the `command` object in this way for later use.
basically you make an assumption that GetServerUrl() always got call first.